Issue 253 upcoming past switch

sergeikofman444 opened the pull request:
## Description

<!-- 1. Link the related issue here. -->

<!-- Replace "Resolves" with "Related to" if this PR should not close the issue (e.g. if the issue requires several PRs to complete) -->
Resolves [issue 253](https://github.com/civic-dashboard/civic-dashboard-web/issues/253)

<!-- 2. Give a high-level description of what this PR does. -->
<!-- What was the behaviour before? What should it be now? -->
This commit replaces the `ShowPastItems` component with a new component, `UpcomingPastSwitch`. It re-uses most of the logic from the previous component, setting the minimum date, but checking if the user is selecting 'past' or 'upcoming' instead of whether the "past items" checkbox is checked or not. 

The new component is still within the search file - I am still getting a hang of how we like to organize things, this seemed like the right place for it but I am happy to discuss. 

<!-- 3. If this PR results in something changing visually on the site, include before/after screenshots. -->
before:
https://github.com/user-attachments/assets/6a2af05c-c9e7-406d-8175-7f1344453379

<img width="1211" height="784" alt="Screenshot 2025-11-24 at 11 33 05 PM" src="https://github.com/user-attachments/assets/9c933ec5-7b29-48c5-b946-250b31a96e15" />

<img width="1146" height="768" alt="Screenshot 2025-11-24 at 11 33 10 PM" src="https://github.com/user-attachments/assets/1272f015-54e5-4b25-ad64-839bf46122f2" />

after: 
https://github.com/user-attachments/assets/0da7418d-7cac-4c5e-bb78-36f8ea1b2a9f

<img width="1012" height="772" alt="Screenshot 2025-11-24 at 11 32 35 PM" src="https://github.com/user-attachments/assets/b9d5f5ad-bf44-445a-9644-14a760c736b0" />

<img width="1175" height="769" alt="Screenshot 2025-11-24 at 11 34 59 PM" src="https://github.com/user-attachments/assets/7bcbcb78-467e-4919-9131-1f38a3fbee00" />

<!-- 4. If you have specific code review directions, include them here. -->
<!-- E.g. "Look at these files first" or "Review one commit at a time" -->

## Testing instructions

<!-- Describe how the reviewer should test this change. -->
<!-- E.g. Which page should they look at? What should they click? What should happen after clicking? -->

- access the actions page (https://e268dc8a-civic-dashboard-web.civic-dashboard.workers.dev/actions)
- toggle between "upcoming items" and "past items"
- confirm the issues are correctly sorted (you may check this against the values in production)
- confirm the UI matches our intended design ([figma](https://www.figma.com/design/edkTzFB3Id1aOg3mH6MGoU/misc-workspace?node-id=0-1&p=f), [issue](https://github.com/civic-dashboard/civic-dashboard-web/issues/253))

## Checklist

<!-- Ensure you do these tasks before requesting a review! -->

- [x] This PR addresses all requirements described in the issue
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my code
- [x] I have tested these changes in the preview deployment
- [x] I have made corresponding changes to the documentation (if relevant)

sergeikofman444 commented on src/components/AgendaItemList.tsx:
```diff
@@ -2,6 +2,7 @@
 
 import { SearchResultAgendaItemCard } from '@/components/AgendaItemCard';
 import {
+  UpcomingPastSwitch,
```
thanks @KarishmaChanglani ! updated

17cliu commented on src/components/search.tsx:
```diff
@@ -74,6 +73,55 @@ export function SortDropdown() {
+            active === 'upcoming'
+              ? 'text-gray-800 border-b-2 border-gray-700'
+              : 'text-gray-400'
+          }`}
```
one thing to watch out for on this project: this site has light _and_ dark modes built-in! the dark mode activates when the os or browser is set to dark mode.

with these styles, the active tab currently looks like it's _not_ the active tab when in dark mode:

<img width="1243" height="1187" alt="Image" src="https://github.com/user-attachments/assets/0e9e574f-df53-46a3-be37-94a0c8643e08" />

you can specify classes for dark mode specifically by prefixing the class with `dark:` -- search for this in the repo to find examples!

sergeikofman444 commented:
thanks @KarishmaChanglani @17cliu ! i've moved the switch to below the search for now. i think this area is still somewhat cluttered but this could be a step on the way while we work on the other stuff related to the filters. @liqidesigns what do you think?

i've also updated it to respect darkmode, and i've added `role="tab"` and `ariaselected` (with respective logic). there may still be some more work to make it fully accessible but lmk what you think in the meantime :) 

<img width="1268" height="595" alt="Screenshot 2025-11-25 at 3 21 16 PM" src="https://github.com/user-attachments/assets/2ccbc1c0-4671-48b3-9a1a-05e307d0bcbb" />

<img width="1268" height="669" alt="Screenshot 2025-11-25 at 3 21 23 PM" src="https://github.com/user-attachments/assets/09f271ef-59b3-4e7d-bf4a-5851c25b855e" />
